composeApp/src/commonMain/kotlin/world/respect/app/app/AppScreenList.kt

```diff
@@ -0,0 +1,45 @@
+package world.respect.app.app
+
+import kotlinx.serialization.Serializable
+import world.respect.datasource.opds.model.OpdsPublication
+
+sealed interface AppDestination
+
+@Serializable
+object AppLauncher : AppDestination
+
+@Serializable
+object Assignment : AppDestination
+
+@Serializable
+object Clazz : AppDestination
+
+@Serializable
+object Report : AppDestination
+
+@Serializable
+object AppList : AppDestination
+
+@Serializable
+object EnterLink : AppDestination
+
+@Serializable
+data class AppsDetail(
+    val manifestUrl: String,
+    val url: String
+) : AppDestination
+
+@Serializable
+data class LessonList(
+    val url: String
+) : AppDestination
+
+@Serializable
+data class LessonDetail(
+    val selfLink: String,
+    val publicationSelfLink: String,
+    val url: String,
```

Comment:
There should be two url parameters (as discussed):

url itself - that is the URL for the lesson manifest ( e.g. lesson001.json as per the test files)
refererUrl (nullable and optional) - the url, if any, for the feed from which it was referred (e.g. the url that was used by LessonList).